ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/ContractStore.scala

```diff
@@ -26,11 +26,30 @@ trait ContractStore {
       loggingContext: LoggingContext
   ): Future[Option[ContractId]]
 
-  /** @return The maximum ledger effective time of all contracts in ids, fails as follows:
-    *         - if ids is empty or not all the non-divulged ids can be found, a failed [[Future]]
-    *         - if all ids are found but each refer to a divulged contract, a successful [[None]]
+  /** This method is serves two purposes:
+    *   1 - Verify that neither of the specified contracts are archived (archival of divulged contracts also count)
+    *   2 - Calculate the maximum ledger time of all the specified contracts (divulged contracts do not contribute)
+    * Important note: existence of the contracts are not checked, only the fact of archival.
+    *
+    * @return NotAvailable, if none of the specified contracts are archived,
```

Comment:
How about moving the scaladoc comments to the `MaximumLedgerTime.*` objects/classes? I have posted some suggestions below.